File: 01568.cpp
#include <iostream>
using namespace std;

int main() {
	int N; cin >> N;

	int sec = 0;
	while (N > 0) {
		int n = 1;
		while ((n * n + n) / 2 <= N) n++;
		n--;

		int an = (n * n + n) / 2;
		sec += n;
		N -= an;
	}

	cout << sec << endl;
}

File: 01568.py
N = int(input())

sec = 0
while N > 0:
    n = 1
    while (n ** 2 + n) // 2 <= N:
        n += 1
    n -= 1

    an = (n ** 2 + n) // 2
    sec += n
    N -= an

print(sec)
